service/api_task.go

```diff
@@ -749,6 +749,29 @@ func assignNextAvailableTask(ctx context.Context, taskQueue *model.TaskQueue, di
 			continue
 		}
 
+		// If the top task on the queue is blocked, the scheduler task queue may be out of date.
+		if nextTask.Blocked() {
+			grip.Debug(message.Fields{
+				"message":            "top task queue task is blocked, dequeuing task",
+				"host_id":            currentHost.Id,
+				"distro_id":          nextTask.DistroId,
+				"task_id":            nextTask.Id,
+				"task_group":         nextTask.TaskGroup,
+				"project":            projectRef.Id,
+				"project_identifier": projectRef.Enabled,
+			})
+			grip.Warning(message.WrapError(taskQueue.DequeueTask(nextTask.Id), message.Fields{
```

Comment:
This dequeue needs to happen per this comment:
https://github.com/evergreen-ci/evergreen/blob/b850ce711b1eacc39cad7ec329f7969ea9c860b1/service/api_task.go#L624-L627
However, it seems like this is only relevant to the legacy dispatcher which isn't in use. Created [EVG-16841](https://jira.mongodb.org/browse/EVG-16841) to address.